### PR TITLE
Hb 04

### PR DESCRIPTION
**Pull request by KostiantynVorobiov**

### CODE REVIEW

**Review by vremere-a — APPROVED** [review]

**Comment by vremere-a** [comment]
Well done

File: src/main/java/com/cinema/model/MovieSession.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+@Entity
+@Table(name = "movie_session")
+public class MovieSession {
+    @Id
+    @Column(name = "session_id")
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne
+    private Movie movie;
+    @ManyToOne
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/model/MovieSession.java:21]: this is for your info, but keep in mind that toOne relations have fetchType eager by default, which is not always quite good

File: src/main/java/com/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.persist(movieSession);
+            transaction.commit();
+            return movieSession;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add movie session", e);
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/dao/impl/MovieSessionDaoImpl.java]: let's make message more informative and add some info about entity we were trying to save same with other methods, where applicable

Comment (KostiantynVorobiov) [reply src/main/java/com/cinema/dao/impl/MovieSessionDaoImpl.java]: Done

File: src/main/java/com/cinema/model/MovieSession.java (hunk 1 above)
Comment (KostiantynVorobiov) [reply src/main/java/com/cinema/model/MovieSession.java:21]: I understand

File: src/main/java/com/cinema/dao/impl/CinemaHallDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.persist(cinemaHall);
+            transaction.commit();
+            return cinemaHall;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add cinema hall", e);
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/dao/impl/CinemaHallDaoImpl.java]: what about this message?

File: src/main/java/com/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> getAllMovieSessionQuery
+                    = session.createQuery("FROM MovieSession WHERE movie.id = :id "
+                    + "AND showTime BETWEEN :start AND :end", MovieSession.class);
+            getAllMovieSessionQuery.setParameter("id", movieId);
+            getAllMovieSessionQuery.setParameter("start", date.atTime(LocalTime.MIN));
+            getAllMovieSessionQuery.setParameter("end", date.atTime(LocalTime.MAX));
+            return getAllMovieSessionQuery.getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get all movie session", e);
```
Comment (RomanDubovskyi) [inline src/main/java/com/cinema/dao/impl/MovieSessionDaoImpl.java]: and this message

File: src/main/java/com/cinema/dao/impl/CinemaHallDaoImpl.java (hunk 3 above)
Comment (KostiantynVorobiov) [reply src/main/java/com/cinema/dao/impl/CinemaHallDaoImpl.java]: oops! Done

File: src/main/java/com/cinema/dao/impl/MovieSessionDaoImpl.java (hunk 4 above)
Comment (KostiantynVorobiov) [reply src/main/java/com/cinema/dao/impl/MovieSessionDaoImpl.java]: I don't know what can I add to this message

Comment (RomanDubovskyi) [reply src/main/java/com/cinema/dao/impl/MovieSessionDaoImpl.java]: > I don't know what can I add to this message

you can add info about input parameters (we're searching sessions for particular movie and date)

Comment (KostiantynVorobiov) [reply src/main/java/com/cinema/dao/impl/MovieSessionDaoImpl.java]: Done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**